### PR TITLE
FT-113 # Replace invalid characters in conditional logic functions

### PR DESCRIPTION
**Pull request by simonmarklar**
- all invalid characters are now removed in the normalisation step
- added tests

### CODE REVIEW

File: packages/forms-transformer/lib/transforms/fix-conditional-logic-names.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+'use strict'
+
+const fixName = (str) => str.replace(/^[0-9]|[^a-z0-9_]/ig, '')
```
Comment (mymattcarroll) [inline packages/forms-transformer/lib/transforms/fix-conditional-logic-names.js]: I hate to be picky but could we please add some comments to explain what this regex is doing.

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,4892 @@
+{
```
Comment (mymattcarroll) [inline package-lock.json]: If we are using `package-lock.json` can we update the `engines` in `package.json` to require 
```json
{
  "npm": ">=5",
}
```

Comment (simonmarklar) [inline package-lock.json]: I dont think we need to update the required version of npm, older versions will just ignore this file, and if we put npm 5 as a requirement we're locking out people who have not updated to a version of node that has npm 5.

When you use npm 5 it automatically writes this file and suggests that you commit it, thats the only reason its there. It doesn't get published to npm either:

https://docs.npmjs.com/files/package-lock.json
>  One key detail about package-lock.json is that it cannot be published

File: packages/forms-transformer/lib/transforms/fix-conditional-logic-names.js (hunk 1 above)
Comment (simonmarklar) [inline packages/forms-transformer/lib/transforms/fix-conditional-logic-names.js]: seems obvious to me but it cant hurt to comment :)

**Comment by mymattcarroll** [comment]
LGTM 👍

**Pull request merged**